fontforge/svg.c

```diff
@@ -1301,14 +1301,18 @@ static void SVGTraceArc(SplineSet *cur,BasePoint *current,
 	y1p =-ang.y*(current->x-x)/2 + ang.x*(current->y-y)/2;
 	/* Correct for bad radii */
 	lambda = x1p*x1p/(rx*rx) + y1p*y1p/(ry*ry);
-	if ( lambda>1 ) {
+	if ( lambda>=1 ) {
 	   lambda = sqrt(lambda);
 	   rx *= lambda;
 	   ry *= lambda;
 	   cxp = cyp = 0;
```

Comment:
Is this line the only one that is operative in the `==` case?

Reply:
Or is the issue that the `else` code causes problems in that case?

Reply:
With the "robot.svg" file:

1. gdb prints `lambda` as exactly 1
2. `factor` before the square root is negative, resulting in NANs.

There's no problem I can see with changing the inequality, so I did that. However, I don't trust that as a general solution (and maybe not even in this case -- gdb print might be rounding), so if the `factor` calculation goes negative I just set it to zero.